include/boost/math/tools/config.hpp

```diff
@@ -470,6 +470,14 @@ namespace boost{ namespace math{
 #else
 #  define BOOST_MATH_THREAD_LOCAL
 #endif
+//
+// Some mingw flavours have issues with thread_local and types with non-trivial destructors
+// See https://sourceforge.net/p/mingw-w64/bugs/527/
+//
```

Comment:
Oh... Now I see... You already have worked out the logic. I could use this in the multiprecision branch for CI...

Reply:
Maybe not: I notice the failing CI run *does* define _REENTRANT so there's still something different going on than on my local machine :(

Reply:
> ... there's still something different going on than on my local machine

An interesting point. The virtual machine on CI runs on `[windows-latest]` which is an alias for Windows Server 2019. Perhaps there is a different C-Library or threading model on the MinGW that somehow got put together on the machine.

It is not immediately clear to me how this might be, but the C runtime library is now shipped as an integral port of a Windows distro, so there might be slight variations in that part...?